### PR TITLE
Fixing performance regression introduced lately

### PR DESCRIPTION
**Pull request by hkaiser**
- hiding extraction of real function address behing PP constant (HPX_WITH_THREAD_DESCRIPTION_FULL)

### CODE REVIEW

File: hpx/apply.hpp [hunk 1]
```diff
@@ -41,7 +42,7 @@ namespace hpx { namespace detail
         >::type
         call(F&& f, Ts&&... ts)
         {
-            util::thread_description desc(f);
+            util::thread_description desc(f, "apply_disptach::call");
```
Comment (sithhell) [inline hpx/apply.hpp]: Small type here.

File: hpx/lcos/dataflow.hpp [hunk 2]
```diff
@@ -205,12 +206,17 @@ namespace hpx { namespace lcos { namespace detail
                 return;
             }
 
+#if defined(HPX_HAVE_THREAD_DESCRIPTION)
+            util::thread_description desc(func_);
+#else
+            util::thread_description desc("dataflow_frame::finalize");
+#endif
```
Comment (sithhell) [inline hpx/lcos/dataflow.hpp]: This should be changed to not use the PP guard.

File: hpx/lcos/local/packaged_continuation.hpp [hunk 3]
```diff
@@ -304,9 +304,13 @@ namespace hpx { namespace lcos { namespace detail
...
             ) = &continuation::async_impl;
 
-            sched.add(
-                util::bind(async_impl_ptr, std::move(this_), f),
-                util::thread_description(f_));
+#if defined(HPX_HAVE_THREAD_DESCRIPTION)
+            util::thread_description desc(f_);
+#else
+            util::thread_description desc("continuation::async");
+#endif
```
Comment (sithhell) [inline hpx/lcos/local/packaged_continuation.hpp]: same here

**Comment by hkaiser** [comment]
The review comment shave been addressed.

**Comment by hkaiser** [comment]
@khuck This patch removes the code which attempts to extract the function addresses (as requested).

**Comment by K-ballo** [comment]
The small typo is still standing

**Comment by hkaiser** [comment]
@K-ballo Thanks, that's fixed now too.

**Comment by K-ballo** [comment]
@hkaiser Look again :)

**Comment by hkaiser** [comment]
How about now? ;)

**Comment by K-ballo** [comment]
Now looks OK, thanks.

**Pull request merged**